### DISCUSSION TITLE
Collection of Evennia code things

### DISCUSSION THREAD

**Discussion by InspectorCaracal** [evennia/evennia#2715] [Community Contribs & Snippets]
I've started to clean up and put some of the more self-contained systems I've been coding into a public repository, in case anyone else wants to use them. Right now it has a room decoration module and an interactive multimatch system, but I have an extension of the Crafting contrib and a weather system in progress as well.

[Here's the repository](https://github.com/InspectorCaracal/evennia-things) and I'll try to remember to post here when I make any major additions or updates.

And here's some brief explanations of the current contents (as of apr 14):

* `basic_commands.py` - Extends the functionality of basic IC commands like `look` and `get`, mostly by adding inter-object functionality. e.g. "get A from B" to get an object from a container object.
* `clothing.py` - Initially based on the current clothing contrib, but rewritten for v1.0 updates and ease of customization. (Don't try backporting this one, just use the current contrib instead.) Like the contrib, it lets you wear objects and displays those objects as part of your character description, with support for covering up worn objects so they can't be seen and optional "worn as" strings appended to the clothing description.
* `growables.py` - Lets your objects change over time. Designed as a flexible handler, it should let you implement anything from growing crops, to aging animals, and even decaying corpses, by changing an object's attributes based on its age.
* `multimatch.py` - Provides an alternative multimatch system to the default of re-entering commands, by allowing commands to ask the player directly which of the matches they intended.
* `roomdecor.py` - Adds a mechanism for decorating rooms with objects. Decor items can be placed in different positions within the room and are displayed as part of room's `desc` section instead of as contents. They also retain any other functionality - usable items are usable, sittable items are sittable, etc.

**Comment by InspectorCaracal** [comment]
I realized I could stick my "extended basics" commands on there, so I did. They add inter-object functionality to key IC commands, like looking at an object that's in another object, or getting an object from another object.

**Comment by InspectorCaracal** [comment]
Added a major overhaul/near-complete rewrite of the official clothing contrib to my repo. I'm angling to put in a PR to replace the current one in develop once I have unit tests, so if any of you try it out and/or have feedback or complaints, lmk here.

**Comment by InspectorCaracal** [comment]
Added a [growth handler](https://github.com/InspectorCaracal/evennia-things/blob/main/growables.py) which I'm actually pretty proud of! It's up there with my room decor in terms of favorite mechanics I've put together so far.

I tried to make it as flexible and agnostic a handler as possible, partly because that's how I roll and partly so I can use it for basically any game object that I want to change over time. I think I succeeded.

**Comment by Griatch** [comment]
Nice to see you post some extra stuff! :+1: 

For people to actually use and comment on any of these things though, I suggest summing up what they do here in the thread before having people click through to the code - from experience I can tell you people will just not do this blind. Making things easy to discover is important. 
What does "extension to the Crafting contrib" mean, for example - without having to dig into your code? What are the improvements to the clothing contrib, in words? A short explanation of what a 'growth handler' is before clicking through? Explanations/examples of what "inter-object functionality in key IC commands" actually _means_ ...

And so on.

**Reply by InspectorCaracal** [reply]
Good idea. I was trying to keep my descriptions from being too wordy but I can see I went too far the other direction.
